poe-tasks/build-and-publish-java-connectors-with-tag.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+# Usage examples:
+#   ./get-modified-connectors.sh --prev-commit --json | ./build-and-publish-java-connectors-with-tag.sh
+#
+# Specific to this script:
+#   1) Default (main-release) on a single connector
+#   ./build-and-publish-java-connectors-with-tag.sh foo-conn
+#   ./build-and-publish-java-connectors-with-tag.sh --name=foo-conn
+#
+#   2) Explicit main-release with multiple connectors
+#   ./build-and-publish-java-connectors-with-tag.sh --publish-option=main-release foo-conn bar-conn
+#
+#   3) Pre-release (dev tag) via JSON pipe
+#   echo '{"connector":["foo-conn","bar-conn"]}' | ./build-and-publish-java-connectors-with-tag.sh --publish-option=pre-release
```

Comment:
nonblocking: I'd mildly prefer to _only_ support this mode, since that's all we need for CI (purely for the sake of having fewer lines of shell script)

Reply:
meaning you prefer only json support?

Reply:
yeah. Unless that's something you're using for local testing/etc

Reply:
yeah pretty much.